instance/instance.go

```diff
@@ -140,20 +165,17 @@ func (h *Instance) ExactImages(images []flux.ImageID) (ImageMap, error) {
 
 // Checks whether the given image exists in the repository.
 // Return true if exist, false otherwise
-func (h *Instance) imageExists(image flux.ImageID) (exists bool, err error) {
-	// Get a list of images
-	images, err := h.registry.GetRepository(image.Repository())
+func (h *Instance) imageExists(image flux.ImageID) (bool, error) {
+	img, err := registry.ParseImage(string(image), nil)
```

Comment:
This is where I was saying the validating `ImageID` stuff probably should go in the `flux` package, so we don't have 2 (subtly) different versions of the same thing.